src/main/java/org/kohsuke/github/GitUser.java

```diff
@@ -27,14 +27,23 @@ public String getName() {
     }
 
     /**
-     * Gets email.
+     * Gets email (e.g. from git commit).
      *
-     * @return E -mail address, such as "foo@example.com"
+     * @return E-mail address, such as "foo@example.com"
      */
     public String getEmail() {
         return email;
     }
 
+    /**
+     * Gets username.
+     *
+     * @return GitHub username
+     */
+    public String getUsername() {
+        return username;
```

Comment:
I don't see this field in the git user information (https://docs.github.com/en/rest/reference/git#get-a-commit).  Where are you finding this?

Reply:
Hm, I see it here: https://github.com/hub4j/github-api/blob/19e1a00fd7c522f94161b153de53accc15b62dfc/src/test/resources/org/kohsuke/github/GHEventPayloadTest/push.json#L21 but it isn't in the current GitHub docs.  Might be only present in older versions of GitHub Server?

Reply:
I have it in GitHub (global; not enterprise) standard webhook notification on `push` event

Reply:
@seregamorph 
Interesting. Is it possible it is only present on events?  
This doesn't include it: https://api.github.com/repos/octocat/Hello-World/git/commits/7fd1a60b01f91b314f59955a4e4d4e80d8edf11d

I'd be okay with adding this if the javadoc clearly states that is only seen on events and we add `@Preview @Deprecated` annotations to it.  I would also ask that you contact GitHub support and ask them to clarify the documentation. 


username

Reply:
Done. Marked as
```
    @Preview
    @Deprecated
    @CheckForNull
```
and added javadoc

Reply:
The topic is published https://github.community/t/push-event-structure-author-committer-id-for-commits/134109 but there is still no answers.